internal/gatewayapi/runner/runner.go

```diff
@@ -182,6 +203,153 @@ func (r *Runner) deleteAllIRKeys() {
 	}
 }
 
+type DeletableStatus struct {
+	GatewayStatusKeys   map[types.NamespacedName]bool
+	HTTPRouteStatusKeys map[types.NamespacedName]bool
+	GRPCRouteStatusKeys map[types.NamespacedName]bool
+	TLSRouteStatusKeys  map[types.NamespacedName]bool
+	TCPRouteStatusKeys  map[types.NamespacedName]bool
+	UDPRouteStatusKeys  map[types.NamespacedName]bool
+
+	ClientTrafficPolicyStatusKeys  map[types.NamespacedName]bool
+	BackendTrafficPolicyStatusKeys map[types.NamespacedName]bool
+	SecurityPolicyStatusKeys       map[types.NamespacedName]bool
+	BackendTLSPolicyStatusKeys     map[types.NamespacedName]bool
+}
+
+func (r *Runner) getDeletableStatus() *DeletableStatus {
```

Comment:
prefer if this API was called `getAllStatuses`

it took me 2 readings to figure out your approach of collecting all statuses and then deleting the current ones from this truck and then deleting the remaining ones from watchable :)